File: vueapp/src/models/windowGlassType.js
import CrudModel from './crudModel.js';
import Brand from './brand.js';
import Image from './image.js';

class WindowGlassType extends CrudModel {
    constructor(json = {}) {
        super(json);
        this.name = json.name || '';
        this.description = json.description || null;
        this.brandId = json.brandId || null;
        this.brand = json.brand ? new Brand(json.brand) : null;
        this.imageId = json.imageId || null;
        this.image = json.image ? new Image(json.image) : null;
        this.thickness = json.thickness || 0.0;
        this.chamberCount = json.chamberCount || 0;
        this.thermalInsulation = json.thermalInsulation || 0.0;
        this.soundInsulation = json.soundInsulation || 0;
        this.antiTheftResistance = json.antiTheftResistance || null;
        this.frosted = json.frosted || false;
    }

	toFormData() {
		const formData = super.toFormData();

		formData.append('name', this.name);
		formData.append('description', this.description);
		formData.append('brandId', this.brandId);
		formData.append('imageId', this.imageId);
		formData.append('thickness', this.thickness);
		formData.append('chamberCount', this.chamberCount);
		formData.append('thermalInsulation', this.thermalInsulation);
		formData.append('soundInsulation', this.soundInsulation);
		formData.append('antiTheftResistance', this.antiTheftResistance);
		formData.append('frosted', this.frosted);

		return formData;
	}

    toSrvData()
    {
        return this.toFormData()
    }
}

export default WindowGlassType;

File: webapi/Models/WindowGlassType.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace webapi.Models
{
    /// <summary>
    /// Representa um tipo de vidro de janela.
    /// </summary>
    public class WindowGlassType : CrudModel
    {
        /// <summary>
        /// Obtém ou define o nome do tipo de vidro da janela.
        /// </summary>
        [Required, MaxLength(255)]
        public string? Name { get; set; }

        /// <summary>
        /// Obtém ou define a descrição do tipo de vidro da janela.
        /// </summary>
        [MaxLength(10000)]
        public string? Description { get; set; }

        [ForeignKey("Brand")]
        public Guid? BrandId { get; set; }

        /// <summary>
        /// Marca
        /// </summary>
        public virtual Brand? Brand { get; set; }
        
        public Guid? ImageId { get; set; }

        /// <summary>
        /// Imagem do vidro
        /// </summary>
        [ForeignKey(nameof(ImageId))]
        public Image? Image { get; set; }

        /// <summary>
        /// Espessura do vidro em milímetros (mm).
        /// </summary>
        public double Thickness { get; set; }

        /// <summary>
        /// Número de camadas da janela.
        /// </summary>
        public int ChamberCount { get; set; } = 0;

        /// <summary>
        /// Isolamento térmico - Uf (W/m²K)
        /// </summary>
        public double ThermalInsulation { get; set; } = 0;

        /// <summary>
        /// Isolamento acústico - Rw (dB)
        /// </summary>
        public int SoundInsulation { get; set; } = 0;

        /// <summary>
        /// Resistência antirroubo
        /// </summary>
        public string? AntiTheftResistance { get; set; }

        /// <summary>
        /// Fosco ?
        /// </summary>
        public bool Frosted { get; set; } = false;

        public WindowGlassType()
            :base()
        {
        }
    }
}
